### DISCUSSION TITLE
Update without removing and readding device

### DISCUSSION THREAD

**Discussion by etkaar** [HomeSpan/HomeSpan#808] [General]
When I update my code and then reupload it to the ESP32-C, it will loose connection to Home Kit. I always need to remove it and then add it again. Is there a method to update it without removing and readding the device?

**Comment by HomeSpan** [comment]
Do you mean that you have to unpair and then re-pair the device with the Home App every time you upload new code?  Are you using a serial connection or OTA?

After a code update, does HomeSpan indicate on the Serial Monitor that it is paired or unpaired when it first boots?

**Comment by etkaar** [comment]
I am using a serial connection. When I recompile and then upload to the ESP32-C3 it says:

`DEVICE NOT YET PAIRED -- PLEASE PAIR WITH HOMEKIT APP`

**Reply by HomeSpan** [reply]
Please check your upload settings.  You may have them set to erase the entire flash before uploading the new code.  If so, this should be disabled otherwise you are erasing the stored pairing info every time you upload new code.

**Reply by etkaar** [reply]
You've been right, that was the reason!
